tests/Compiler/string-globals.j

```diff
@@ -0,0 +1,25 @@
+; RUN: jasmin %s -d %t
+; RUN: jllvm-jvmc --method "test:()V" %t/Test.class | FileCheck %s
+
+.class public Test
+.super java/lang/Object
+
+.method public <init>()V
+    aload_0
+    invokespecial java/lang/Object/<init>()V
+    return
+.end method
+
+.method public static native print(Ljava/lang/String;)V
+.end method
+
+; CHECK-LABEL: define void @"Test.test:()V"
+.method public static test()V
+    .limit stack 1
+    ; CHECK: store ptr addrspace(1) @"'Test String", ptr %[[TOP:[[:alnum:]]+]]
+    ldc "Test String"
+    ; CHECK-NEXT: %[[ARG:[[:alnum:]]+]] = load ptr addrspace(1), ptr %[[TOP]],
+    ; CHECK-NEXT: call void @"Static Call to Test.print:(Ljava/lang/String;)V"(ptr addrspace(1) %[[ARG]])
```

Comment:
I would remove these two check lines as they are not testing the new string interning mechanism per-se. The thing really being tested here is the `ldc` instruction which you do nicely above.